commonItems/ParserHelpers.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+using System.Numerics;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace commonItems {
+    public static class ParserHelpers {
+        public static void IgnoreItem(BufferedReader sr) {
+            var next = Parser.GetNextLexeme(sr);
+            if (next == "=") {
+                next = Parser.GetNextLexeme(sr);
+            }
+            if (next is "rgb" or "hsv") // Needed for ignoring color. Example: "color = rgb { 2 4 8 }"
+            {
+                if ((char)sr.Peek() == '{') {
+                    next = Parser.GetNextLexeme(sr);
+                } else { // don't go further in cases like "type = rgb"
+                    return;
+                }
+            }
+            if (next == "{") {
+                var braceDepth = 1;
+                while (true) {
+                    if (sr.EndOfStream) {
+                        return;
+                    }
+                    var token = Parser.GetNextLexeme(sr);
+                    switch (token) {
+                        case "{":
+                            ++braceDepth;
+                            break;
+                        case "}": {
+                                --braceDepth;
+                                if (braceDepth == 0) {
+                                    return;
+                                }
+                                break;
+                            }
+                    }
+                }
+            }
+        }
+
+        public static void IgnoreAndLogItem(BufferedReader sr, string keyword) {
+            IgnoreItem(sr);
+            Logger.Log(LogLevel.Debug, "Ignoring keyword: " + keyword);
+        }
+    }
+    public class SingleString : Parser {
+        public SingleString(BufferedReader sr) {
+            GetNextTokenWithoutMatching(sr); // remove equals
```

Comment:
As elsewhere, consider putting the comment just ahead of the line of code it's about.